modules/planning/tasks/optimizers/open_space_trajectory_partition/open_space_trajectory_partition.cc

```diff
@@ -441,26 +442,30 @@ void OpenSpaceTrajectoryPartition::PartitionTrajectory(
             : canbus::Chassis::GEAR_REVERSE;
 
     if (cur_gear != *gear) {
+      is_trajectory_last_point = true;
+      LoadTrajectoryPoint(trajectory_point, is_trajectory_last_point, *gear,
+                          &last_pos_vec, &distance_s, trajectory);
       partitioned_trajectories->emplace_back();
       current_trajectory_gear = &(partitioned_trajectories->back());
       current_trajectory_gear->second = cur_gear;
       distance_s = 0.0;
+      is_trajectory_last_point = false;
     }
 
     trajectory = &(current_trajectory_gear->first);
     gear = &(current_trajectory_gear->second);
 
-    LoadTrajectoryPoint(trajectory_point, *gear, &last_pos_vec, &distance_s,
-                        trajectory);
+    LoadTrajectoryPoint(trajectory_point, is_trajectory_last_point, *gear,
+                        &last_pos_vec, &distance_s, trajectory);
   }
-
+  is_trajectory_last_point = true;
   const TrajectoryPoint& last_trajectory_point = raw_trajectory.back();
-  LoadTrajectoryPoint(last_trajectory_point, *gear, &last_pos_vec, &distance_s,
-                      trajectory);
+  LoadTrajectoryPoint(last_trajectory_point, is_trajectory_last_point, *gear,
+                      &last_pos_vec, &distance_s, trajectory);
 }
 
 void OpenSpaceTrajectoryPartition::LoadTrajectoryPoint(
-    const TrajectoryPoint& trajectory_point,
+    const TrajectoryPoint& trajectory_point, bool is_trajectory_last_point,
```

Comment:
use const bool is_trajectory_last_point?

Reply:
My bad, it is been fixed. Thanks.